libraries/app/api.cpp

```diff
@@ -530,16 +530,17 @@ namespace graphene { namespace app {
     }
 
     // asset_api
-    asset_api::asset_api(application& app) :_app(app),_db(std::ref(*app.chain_database())) {
-	    FC_ASSERT(_app.chain_database() );
-    }
+    asset_api::asset_api(graphene::app::application& app) :
+         _app(app),
+         _db( *app.chain_database()),
+         database_api( std::ref(*app.chain_database()), &(app.get_options())
+         ) { }
     asset_api::~asset_api() { }
 
-    vector<account_asset_balance> asset_api::get_asset_holders( asset_id_type asset_id, uint32_t start, uint32_t limit ) const {
-
+    vector<account_asset_balance> asset_api::get_asset_holders( std::string asset, uint32_t start, uint32_t limit ) const {
```

Comment:
why?

Reply:
This change is already in `develop` branch. https://github.com/bitshares/bitshares-core/pull/1272

Reply:
t was added after I made the updates for the PR. had to merge the changes for the develop branch

Reply:
This should be handled during the merge, not in advance.

Reply:
@pmconrad IIRC there was a conflict, fixed by merging develop branch into this branch. IMHO it's the correct approach. Actually, if you check [the whole PR diff](https://github.com/bitshares/bitshares-core/pull/1513/files) or [the "condensed" merge commit](https://github.com/bitshares/bitshares-core/pull/1513/commits/adc0086ce87dc4b2e5327047080572ae8b142775) but not [the full merge commit](https://github.com/bitshares/bitshares-core/commit/adc0086ce87dc4b2e5327047080572ae8b142775), you won't see this change.

Reply:
Merge conflicts should be fixed during the merge.

In this case, some code was c&p'ed from develop, maybe in order to avoid a merge conflict. That's the wrong approach.

I know the accidentally added / removed code doesn't show up in the full PR diff, but I'd like to have a clean and lean history. That's why `git rebase -i` exists.